### ISSUE TITLE
language-ecmascript 0.17 & 0.16.2 build failure on GHC 7.10

### ISSUE THREAD

**Issue by bergmark** [haskell-infra/hackage-trustees#23]
Same build error for both versions

```
[ 5 of 14] Compiling Language.ECMAScript3.PrettyPrint ( src/Language/ECMAScript3/PrettyPrint.hs, dist/dist-sandbox-8f2c37ec/build/Language/ECMAScript3/PrettyPrint.o )

src/Language/ECMAScript3/PrettyPrint.hs:50:54:
    Ambiguous occurrence ‘<$>’
    It could refer to either ‘Text.PrettyPrint.Leijen.<$>’,
                             imported from ‘Text.PrettyPrint.Leijen’ at src/Language/ECMAScript3/PrettyPrint.hs:12:1-46
                          or ‘Prelude.<$>’,
                             imported from ‘Prelude’ at src/Language/ECMAScript3/PrettyPrint.hs:14:1-33
                             (and originally defined in ‘Data.Functor’)
```

Fixed by restricting to `base < 4.8`, no further action necessary.

Pinging @achudnov.

**Comment by achudnov** [comment]
There is a candidate fix, but I'll need a few days to set up an environment with 7.10 to test it.

**Comment by achudnov** [comment]
Fix released as 0.17.0.1 (https://hackage.haskell.org/package/language-ecmascript-0.17.0.1)

**Comment by hvr** [comment]
To make this complete, I also added a few lower/upper bounds to the historic versions `0.10` through `0.15.2`, specifically:

 - `QuickCheck` from `>=2.5` to `>=2.5 && <2.7`
 - `template-haskell` from `== *` to `>=2.6`
 - `containers` from `>=0.1` to `>=0.4.2`

This mostly improves the experience for GHC 7.0/7.2/7.4 users
